### PR TITLE
Greg/localnet

### PR DESCRIPTION
**Pull request by greg-szabo**
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -178,6 +178,23 @@ metalinter_all:
...
+### Local testnet using docker - for developer use only!
+
+# Build linux binary on other platforms
+build-linux:
+	GOOS=linux GOARCH=amd64 $(MAKE) build
+
+# Run a 4-node testnet locally
+docker-start:
+#	@echo "Wait until 'Attaching to mach0, mach1, mach2, mach3' message appears"
+	@if ! [ -f build/mach0/config/genesis.json ]; then docker run --rm -e LOG="stdout" -v `pwd`/build:/tendermint tendermint/localnode testnet --dir . --n 4 ; fi
```
Comment (greg-szabo) [inline Makefile]: format was changed

File: docker-compose/Makefile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Makefile for the "localnode" docker image.
+
+localnode:
```
Comment (greg-szabo) [inline docker-compose/Makefile]: rename to all

File: docker-compose/localnode/wrapper.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	echo "The binary `basename ${BINARY}` cannot be found. Please add the binary to the shared folder. E.g.: -e BINARY=tendermint_my_test_version"
+	exit 1
+fi
+BINARY_CHECK="`file $BINARY | grep 'ELF 64-bit LSB executable, x86-64'`"
+if [ -z "${BINARY_CHECK}" ]; then
+	echo "Binary needs to be OS linux, ARCH amd64"
+	exit 1
+fi
+
+##
```
Comment (greg-szabo) [inline docker-compose/localnode/wrapper.sh]: remove

File: docker-compose/localnode/wrapper.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+BINARY=/tendermint/${BINARY:-tendermint}
+ID=${ID:-0}
+LOG=${LOG:-tendermint.log}
+
+##
+## Assert linux binary
+##
+if ! [ -f "${BINARY}" ]; then
+	echo "The binary `basename ${BINARY}` cannot be found. Please add the binary to the shared folder. E.g.: -e BINARY=tendermint_my_test_version"
```
Comment (greg-szabo) [inline docker-compose/localnode/wrapper.sh]: `. E.g.:` shared folder or use env variable to set binary

**Comment by melekes** [comment]
please rebase against latest develop

**Comment by greg-szabo** [comment]
I don't have access to merge this.

File: Makefile [hunk 5]
```diff
@@ -178,6 +178,23 @@ metalinter_all:
 	@echo "--> Running linter (all)"
 	gometalinter.v2 --vendor --deadline=600s --enable-all --disable=lll ./...
 
+###########################################################
+### Local testnet using docker
+
+# Build linux binary on other platforms
+build-linux:
```
Comment (melekes) [inline Makefile]: can you add these commands to `.PHONY`

Comment (greg-szabo) [reply Makefile]: yes, good catch!

**Pull request merged**